### ISSUE TITLE
Multiple spell casts per turn

### ISSUE THREAD

**Issue by thekeatonfox** [ZerdBG3/DnD-Rebalancing#35]
Hi, Mithra here!
As stated on discord at #suggestions

>playing today I realized I could cast a healing word as a bonus action and a guiding bolt as an action at the same turn. If i'm not mistaken, you could only cast cantrips in this scenario where you used your bonus action to cast a spell.

https://5e.tools/book.html#phb,10,bonus%20action,0

Thanks,

**Comment by gnufred** [comment]
@thekeatonfox You're not mistaken. Official rules, can't cast two spells in the same turn, unless one is a cantrip.

**Comment by Llakar** [comment]
I believe the rules are based on if you cast a bonus action, your other spell has to be a cantrip. But you can cast two non-cantrip spells with action followed by action surge. This is what made the issue with counterspell by RAW. Say a wizard casts an action spell and another spell caster counterspells. The wizard can counterspell that. However, if the wizard had used a bonus action spell, they would not be able to counterspell the counterspell. A lot of people don't like that and ignore this part of the rule.

**Comment by FelipeRenault** [comment]
Larian added some new things for Sorcerer. I'll keep note here on how I think it's possible to solve this issue
- `HasUseCosts('BonusActionPoint')` allows checking for a spell cost. Implement a Passive that applies an `ActionResourceBlock` for `SpellSlot` for the rest of the turn when a spell with that use cost is cast, and give that Passive to everyone.

Now, that only solves the issue of casting a spell as a bonus action then trying to cast a levelled spell afterwards, it can't block someone from casting a levelled spell first with their action and then casting a levelled spell as a bonus action. I need to check if there's any other way to block spell usage by checking their action cost proactively
